genanki/note.py

```diff
@@ -27,6 +28,30 @@ def sort_field(self, val):
   # __init__ and it'll still work.
   @cached_property
   def cards(self):
+    if self.model.model_type == self.model.FRONT_BACK:
+      return self._front_back_cards()
+    return self._cloze_cards()
+
+  def _cloze_cards(self):
+    """returns a Card with unique ord for each unique cloze reference"""
+    if self.model.model_type != self.model.CLOZE:
+      raise RuntimeError('Expected model_type CLOZE or FRONT_BACK')
```

Comment:
`ValueError` would be better than `RuntimeError` here. That's the preferred exception to use when an argument passed to a function does not have a valid value.

For example, Python raises a `ValueError` when you try to do `math.sqrt(-1)`:
```
In [1]: import math                                                                                                                                                                                                                         

In [2]: math.sqrt(-1)                                                                                                                                                                                                                       
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-2037b8d41d70> in <module>
----> 1 math.sqrt(-1)

ValueError: math domain error
```

Reply:
Good idea. Done: https://github.com/kerrickstaley/genanki/pull/37#discussion_r333657905